lua/plenary/async/control.lua

```diff
@@ -207,7 +218,7 @@ M.channel.mpsc = function()
     if deque:is_empty() then
       condvar:wait()
     end
-    local val = deque:popright()
+    local val = deque:popleft()
```

Comment:
it's not very clear to make what `Receiver.last` does.

I assume it was supposed to return the last pushed item? If so, then this change should fix it.

Let me know if this is the correct understanding.

Reply:
yeah, that's the right understanding. Last should always return the most recently sent thing. I think this makes sense.